crates/driver/src/tests/setup/solver.rs

```diff
@@ -1,63 +1,235 @@
-use std::{
-    net::SocketAddr,
-    sync::{Arc, Mutex},
+use {
+    super::{blockchain, blockchain::Blockchain},
+    crate::{
+        domain::competition::{auction, order},
+        infra::{self, blockchain::contracts::Addresses, Ethereum},
+        tests::hex_address,
+    },
+    itertools::Itertools,
+    serde_json::json,
+    std::{
+        collections::{HashMap, HashSet},
+        net::SocketAddr,
+        sync::{Arc, Mutex},
+    },
 };
 
-/// Configuration for mocking a solver.
-#[derive(Debug, Default)]
-pub struct Config {
-    /// The auctions and solutions that the solver should expect and respond
-    /// with.
-    pub solve: Vec<Solve>,
-    pub name: String,
-    pub absolute_slippage: String,
-    pub relative_slippage: String,
-    pub address: String,
-    pub private_key: String,
-}
+pub const NAME: &str = "test-solver";
 
-#[derive(Debug, Clone)]
-pub struct Solve {
-    pub req: serde_json::Value,
-    pub res: serde_json::Value,
+pub struct Solver {
+    pub addr: SocketAddr,
 }
 
-/// A solver that was started as part of the setup.
 #[derive(Debug)]
-pub struct Solver {
-    pub config: Config,
-    pub addr: SocketAddr,
+pub struct Config<'a> {
+    pub blockchain: &'a Blockchain,
+    pub solutions: &'a [blockchain::Solution],
+    pub trusted: &'a HashSet<&'static str>,
+    pub quotes: &'a [super::blockchain::Quote],
+    pub deadline: chrono::DateTime<chrono::Utc>,
+    pub now: infra::time::Now,
+    /// Is this a test for the /quote endpoint?
+    pub quote: bool,
 }
 
-/// Set up an HTTP server exposing a solver API and acting as a solver mock.
-pub async fn setup(config: Config) -> Solver {
-    let state = Arc::new(Mutex::new(config.solve.clone()));
-    let app = axum::Router::new()
+impl Solver {
+    /// Set up an HTTP server exposing a solver API and acting as a solver mock.
+    pub async fn new(config: Config<'_>) -> Self {
+        let mut solutions_json = Vec::new();
+        let mut orders_json = Vec::new();
+        for quote in config.quotes {
```

Comment:
If I understood the `Test` code correctly `quote` tests are mutually exclusive with `solve` tests so it would make sense to me to either build responses for `quotes` or `solutions` here.

Reply:
This is one of the more messy parts of the setup framework. These _quotes_ are `setup::blockchain::Quote`s, which simply represent the achievable prices. They have nothing to do with the driver quoting process. :grimacing: